### PR TITLE
Protocol skip fixes

### PR DESCRIPTION
**Pull request by mambocab**
Addresses the issues brought up in #498. @iamaleksey can you review to double-check the protocol version skips?

This also makes some formatting fixes and removes unused variables. @ptnapoleon can you review those?

### CODE REVIEW

File: write_failures_test.py [hunk 1]
```diff
@@ -169,32 +169,31 @@ def test_counter(self):
...
-    @since('2.2')
+    @since('2.0')
     def test_thrift(self):
         """
-            A thrift client receives a TimedOutException
+        A thrift client receives a TimedOutException
         """
         self.expected_expt = thrift_types.TimedOutException
 
-        session = self._prepare_cluster(start_rpc=True)
```
Comment (ptnapoleon) [inline write_failures_test.py]: I am fairly sure you need to call this.

Comment (mambocab) [inline write_failures_test.py]: facepalm.jpg

**Comment by mambocab** [comment]
The _v2 and _v3 tests fail (expecting but not seeing a `WriteTimeout`), as well as `test_thrift` (expecting but not seeing a `TimedOutException`), on Cassandra 2.1. @iamaleksey Can you have a look and let me know if that's expected. If you like, I can file a Jira and find someone else to have a look.

**Comment by ptnapoleon** [comment]
Bump

**Comment by ptnapoleon** [comment]
Bump.

**Comment by mambocab** [comment]
Ok, this is ready for review. @stef1927 or @iamaleksey, can one of you review please?

I still get the following failure on `write_failures_test.py:TestWriteFailures.test_thrift` on :

```
======================================================================
FAIL: write_failures_test.py:TestWriteFailures.test_thrift
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mambocab/cstar_src/cassandra-dtest/tools.py", line 253, in wrapped
    f(obj)
  File "/home/mambocab/cstar_src/cassandra-dtest/write_failures_test.py", line 201, in test_thrift
    thrift_types.ConsistencyLevel.ALL)
AssertionError: TimedOutException not raised
-------------------- >> begin captured logging << --------------------
dtest: DEBUG: cluster ccm directory: /tmp/dtest-Y1SwpZ
--------------------- >> end captured logging << ---------------------
```

What is the correct behavior for this test?

File: write_failures_test.py [hunk 2]
```diff
@@ -80,31 +80,31 @@ def _perform_cql_statement(self, text):
...
 
-    @since('2.2', '2.2.X')
+    @since('2.0', '2.2.X')
     def test_mutation_v2(self):
         """
-            A failed mutation at v2 receives a WriteTimeout
+        A failed mutation at v2 receives a WriteTimeout
         """
-        self.expected_expt = WriteTimeout
+        self.expected_expt = (WriteTimeout if self.cluster.version() >= '2.2' else None)
```
Comment (mambocab) [inline write_failures_test.py]: I am not confident that this behavior is correct; I chose to unskip it based on [this comment from @stef1927](https://github.com/riptano/cassandra-dtest/pull/498#issuecomment-134416772).

File: write_failures_test.py [hunk 3]
```diff
@@ -80,31 +80,31 @@ def _perform_cql_statement(self, text):
...
 
-    @since('2.2')
+    @since('2.1')
     def test_mutation_v3(self):
         """
-            A failed mutation at v3 receives a WriteTimeout
+        A failed mutation at v3 receives a WriteTimeout
         """
-        self.expected_expt = WriteTimeout
+        self.expected_expt = (WriteTimeout if self.cluster.version() >= '2.2' else None)
```
Comment (mambocab) [inline write_failures_test.py]: Same as above.

**Comment by iamaleksey** [comment]
Don't really remember, sorry. Hopefully @stef1927 will help out.

**Comment by stef1927** [comment]
The 2.1 tests are not failing because in `_prepare_cluster()` we rely on this flag that is only implemented in 2.2: `-Dcassandra.test.fail_writes_ks`. I must have forgotten this when I wrote [this comment](https://github.com/riptano/cassandra-dtest/pull/498#issuecomment-134416772).

So I would keep the `@since('2.2')` tags and add some documentation for that flag.

The rest of the patch is +1.

**Comment by mambocab** [comment]
@stef1927 Looks like all the test cases either call `_prepare_cluster` or `_perform_cql_statement`, which calls `_prepare_cluster`, so all the tests here shouldn't work pre-2.2 -- can you confirm that's what you meant?

In light of that I skipped all the tests at the class level when running on pre-2.2 C*. If I get a +1 I'll rebase and merge.

**Comment by stef1927** [comment]
+1, all tests will not work before 2.2 because the test flag was only implemented in 2.2.

**Comment by mambocab** [comment]
Did a little cleanup rebasing and commenting, merging now. Thanks everyone!

**Pull request merged**